src/machine/machine_nrf528xx.go

```diff
@@ -0,0 +1,102 @@
+//go:build nrf52840 || nrf52833
+
+package machine
+
+import (
+	"device/nrf"
+	"unsafe"
+)
+
+// I2C on the NRF528xx.
+type I2C struct {
+	Bus *nrf.TWIM_Type
+}
+
+// There are 2 I2C interfaces on the NRF.
+var (
+	I2C0 = &I2C{Bus: nrf.TWIM0}
+	I2C1 = &I2C{Bus: nrf.TWIM1}
+)
+
+func (i2c *I2C) enableAsController() {
+	i2c.Bus.ENABLE.Set(nrf.TWIM_ENABLE_ENABLE_Enabled)
+}
+
+func (i2c *I2C) disable() {
```

Comment:
Perhaps we should export this method? I think it might make it easier later for low power modes on other processors.

Reply:
Actually, we can just leave it as it is for now in your implementation. If we export something, people will automatically expect it, and we probably need to do some additional work to do what I was thinking.